langchain/llms/custom_endpoint.py

```diff
@@ -0,0 +1,133 @@
+"""Wrapper around custom APIs."""
+from typing import Any, Dict, List, Mapping, Optional
+
+import requests
+import asyncio
+from functools import partial
+from pydantic import Extra, root_validator
+
+from langchain.llms.base import LLM
+from langchain.llms.utils import enforce_stop_tokens
+from langchain.utils import get_from_dict_or_env
+
+VALID_TASKS = ("text2text-generation", "text-generation")
+
+
+class CustomEndpoint(LLM):
+    """Wrapper around custom Inference Endpoints, related to HuggingFaceHub Inference Endpoints.
+
+    Only supports `text-generation` and `text2text-generation` for now.
+
+    Example:
+        .. code-block:: python
+
+            from langchain.llms import CustomEndpoint
+            endpoint_url = "https://api/endpoint/
+            llm = CustomEndpoint(
+                endpoint_url=endpoint_url,
+                api_key="my-api-key"
+            )
+    """
+
+    endpoint_url: str = ""
```

Comment:
Is there ever a case where an empty string would be valid? It seems like an argument that shouldn't have a default. @hwchase17 looks like it's copied from the HuggingFaceEndpoint one

Reply:
hmmm i dont remember

Reply:
It's not valid; an error will be raised. So it definitely shouldn't be optional.